### PR TITLE
refactored database and views

### PR DESCRIPTION
**Pull request by vincecloutier**
Also, now the presenters only depend on View - which is an abstract class (while not an interface, I think it's better than depending on concrete class that itself depends on everything else like we did before). I'll speak to TA today about it and if there's a better alternative we'll run with that.

### CODE REVIEW

File: src/main/java/database/UserDBAccess.java [hunk 1]
```diff
@@ -1,106 +1,55 @@
...
-            JsonArray usersArray = jsonObject.get("users").getAsJsonArray();
-            List<UserDBModel> users = new ArrayList<>();
-            for (JsonElement userElement : usersArray) {
-                JsonObject userObject = userElement.getAsJsonObject();
-                int userId = userObject.get("id").getAsInt();
-                String username = userObject.get("username").getAsString();
-                String password = userObject.get("password").getAsString();
-                users.add(new UserDBModel(userId, username, password));
-            }
+    public List<UserDBModel> retrieveEveryUser() {
```
Comment (aldew5) [inline src/main/java/database/UserDBAccess.java:15]: nitpick but I think fetchUsers or getUsers is more standard terminology

**Review by aldew5 — COMMENTED** [review]
I'm not sure if you should move Main outside the default package. I think maybe move everything in Main except the main function into View. Then you instantiate a view wherever you have Main now

File: src/main/java/database/UserDBAccess.java (hunk 1 above)
Comment (vincecloutier) [reply src/main/java/database/UserDBAccess.java]: see f447606

File: src/main/java/Main.java [hunk 2]
```diff
@@ -1,32 +1,33 @@
...
         RoomDBGateway roomDBAccess = new RoomDBAccess(API_URL);
 
-        UserPresenter userPresenter = new UserPresenter(roomDBAccess);
+        UserPresenter userPresenter = new UserPresenter();
         UserInteractor userInteractor = new UserInteractor(userPresenter, userDBAccess);
         UserController userController = new UserController(userInteractor);
         WelcomeView welcomeView = new WelcomeView(userController);
-        welcomeView.prepareGUI();
 
+        LeaveRoomPresenter leaveRoomPresenter = new LeaveRoomPresenter();
```
Comment (chelkisea) [inline src/main/java/Main.java:18]: not related to DB stuffs but do you think it is worth keeping left interface right class rule (what we learned in class) here?

File: src/main/java/host_room/HostRoomPresenter.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package host_room;
+
+import ui.View;
+
+import javax.swing.JOptionPane;
+public class HostRoomPresenter implements HostRoomOutputBoundary {
+    private View view;
```
Comment (chelkisea) [inline src/main/java/host_room/HostRoomPresenter.java:7]: is it okay to directly initialize abstract class?

Comment (vincecloutier) [reply src/main/java/host_room/HostRoomPresenter.java:7]: I don't think we're doing that here? We initialize the children of `View` in our `main` method and pass those in here.

File: src/main/java/Main.java [hunk 4]
```diff
@@ -1,32 +1,33 @@
...
+        LeaveRoomInteractor leaveRoomInteractor = new LeaveRoomInteractor(roomDBAccess, leaveRoomPresenter);
+        LeaveRoomController leaveRoomController = new LeaveRoomController(leaveRoomInteractor);
+        RoomView roomView = new RoomView(leaveRoomController);
+
+        HostRoomPresenter hostRoomPresenter = new HostRoomPresenter();
+        HostRoomInteractor hostRoomInteractor = new HostRoomInteractor(roomDBAccess, hostRoomPresenter);
```
Comment (chelkisea) [inline src/main/java/Main.java:24]: not related to DB stuffs but do you think it is worth keeping left interface right class rule (what we learned in class) here?

Comment (vincecloutier) [reply src/main/java/Main.java:24]: I don't think that's a hard and fast rule and if we do that we'll have to cast down the line like so: <br> `((UserPresenter) userPresenter).setView(joinOrHostView);` <br> Let me know if you think it's still worth doing.

**Review by chelkisea — APPROVED** [review]
I think this is great and I agree that using abstract class is better than using interface for View. So we can reduce unnecessary code duplication.

**Review by margaretpell — APPROVED** [review]

**Review by weihan1 — APPROVED** [review]

**Pull request merged**